sentry-spring/src/main/java/io/sentry/spring/tracing/SentryTracingFilter.java

```diff
@@ -67,16 +67,22 @@ protected void doFilterInternal(
       final ITransaction transaction = startTransaction(httpRequest, sentryTraceHeader);
       try {
         filterChain.doFilter(httpRequest, httpResponse);
+      } catch (Exception e) {
+        // exceptions that are not handled by Spring
+        transaction.setStatus(SpanStatus.INTERNAL_ERROR);
       } finally {
         // after all filters run, templated path pattern is available in request attribute
         final String transactionName = transactionNameProvider.provideTransactionName(httpRequest);
         // if transaction name is not resolved, the request has not been processed by a controller
-        // and
-        // we should not report it to Sentry
+        // and we should not report it to Sentry
         if (transactionName != null) {
           transaction.setName(transactionName);
           transaction.setOperation(TRANSACTION_OP);
-          transaction.setStatus(SpanStatus.fromHttpStatusCode(httpResponse.getStatus()));
+          // if exception has been thrown, transaction status is already set to INTERNAL_ERROR, and
+          // httpResponse.getStatus() returns 200.
+          if (transaction.getStatus() == null) {
+            transaction.setStatus(SpanStatus.fromHttpStatusCode(httpResponse.getStatus()));
```

Comment:
pass a sensible default? okhttp uses `SpanStatus.INTERNAL_ERROR`

Reply:
I am not sure if this is right. All 3xx, 422 and more responses would get INTERNAL_ERROR

Reply:
better than no status at all?

Reply:
https://github.com/getsentry/develop/issues/307

Reply:
could be also `UNKNOWN`, but I'd rather change it everywhere to `UNKNOWN`

Reply:
now its `UNKNOWN` if not a match, but other integrations do `INTERNAL_ERROR`, so I'd either keep this one as `INTERNAL_ERROR` or change all the others to use `UNKNOWN` as a default, I'd not like to differ here within the same SDK.

Reply:
we recently had a discussion about that (during a call), cc @bruno-garcia 
we might not set any status actually because it'll fall back on the server side, so then we should remove any default I believe, wdyt @bruno-garcia

Reply:
agree no status is better than possibly misleading status.

Reply:
Reverted the change to not set status.